File: website_front/src/types/reward.ts
export type RewardStatus = 'AVAILABLE' | 'REDEEMED' | 'EXPIRED';

export interface Reward {
  id: number;
  userId: number;
  title: string;
  description?: string;
  imageUrl?: string;
  validUntil: string;
  status: RewardStatus;
  createdAt: string;
  redeemedAt?: string;
  notificationHistoryId?: number;
}

export interface UserNotification {
  id: number;
  userId?: number;
  title: string;
  body: string;
  imageUrl?: string;
  deeplink?: string;
  createdAt: string;
  readAt?: string;
  source: string;
  payloadJson?: string;
}

export interface Page<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

File: website_back/src/main/java/com/baronesa/website/entity/Reward.java
package com.baronesa.website.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "rewards")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reward {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(length = 500)
    private String imageUrl;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDateTime validUntil;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private RewardStatus status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime redeemedAt;

    @Column(name = "notification_history_id")
    private Long notificationHistoryId;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}